lib/geojson.js

```diff
@@ -22,6 +29,99 @@ export function addressesToGeoJson(addresses) {
   }
 }
 
+export function voieNumerosToGeoJson(numeros) {
+  const numerosWithPos = numeros.filter(numero => numero.positions.length > 0)
+
+  return {
+    type: 'FeatureCollection',
+    features: numerosWithPos.map(numero => {
+      const positions = numero.edited ? numero.modified.positions : numero.positions
+      const position = positions[0]
+
+      return {
+        type: 'Feature',
+        geometry: {
+          type: 'Point',
+          coordinates: position.coords || position.coordinates
+        },
+        properties: {
+          ...numero
+        }
+      }
+    })
+  }
+}
+
+export function communeVoiesToGeoJson(commune) {
+  const voies = Object.keys(commune.voies).map(voies => commune.voies[voies])
+  const filteredVoies = voies.filter(voie => voie.numeros)
+
+  return {
+    type: 'FeatureCollection',
+    features: filteredVoies.map(voie => {
+      const numeros = Object.keys(voie.numeros).map(n => voie.numeros[n])
+      const numeroFeatures = voieNumerosToGeoJson(numeros).features
+      const centroid = createCentroid(numeroFeatures)
+
+      return {
+        type: 'Feature',
+        geometry: centroid.geometry,
+        properties: {
+          codeCommune: commune.code,
+          codeVoie: voie.codeVoie,
+          nomVoie: voie.modified && voie.modified.nomVoie ? voie.modified.nomVoie : voie.nomVoie,
+          status: getStatus(voie),
+          numerosCount: String(numeroFeatures.length)
+        }
+      }
+    })
+  }
+}
+
+export function communeNumerosToGeoJson(commune) {
+  const geojson = {
+    type: 'FeatureCollection',
+    features: []
+  }
+
+  Object.keys(commune.voies).forEach(voieIdx => {
+    const voie = commune.voies[voieIdx]
+    if (voie.numeros) {
+      Object.keys(voie.numeros).forEach(numeroIdx => {
+        const numero = commune.voies[voieIdx].numeros[numeroIdx]
+        const positions = numero.edited ? numero.modified.positions : numero.positions
+
+        if (positions.length > 0) {
+          geojson.features.push({
+            type: 'Feature',
+            geometry: {
+              type: 'Point',
+              coordinates: positions[0].coords
+            },
+            properties: {
+              ...numero,
+              codeCommune: commune.code,
+              codeVoie: voie.codeVoie,
+              nomVoie: voie.nomVoie,
+              source: positions[0].source,
+              type: positions[0].type,
+              status: getStatus(numero),
+              voieStatus: getStatus(voie),
+              lastUpdate: positions[0].dateMAJ,
+              color: randomColor({
+                seed: voie.codeVoie,
+                luminosity: 'dark'
+              })
+            }
+          })
+        }
+      })
+    }
+  })
```

Comment:
```js
for (const voie of Object.values(commune.voies)) {
  if (voie.numeros) {
    for (const numero of Object.values(voie.numeros)) {
      const positions = numero.edited ? numero.modified.positions : numero.positions
      if (positions.length > 0) {
        geojson.features.push({
          type: 'Feature',
          geometry: {
            type: 'Point',
            coordinates: positions[0].coords
          },
          properties: {
            ...numero,
            codeCommune: commune.code,
            codeVoie: voie.codeVoie,
            nomVoie: voie.nomVoie,
            source: positions[0].source,
            type: positions[0].type,
            status: getStatus(numero),
            voieStatus: getStatus(voie),
            lastUpdate: positions[0].dateMAJ,
            color: randomColor({
              seed: voie.codeVoie,
              luminosity: 'dark'
            })
          }
        })
      }
    }
  }
}
```

Untested.

Reply:
Why use the indexed form of `@etalab/bal` if you’re only using arrays?

Reply:
Afin d'être manipulé dans `model.js` grâce aux codes commune/voie/numéro.